src/Rendering.md

```diff
@@ -1,85 +1,73 @@
-rectangle and only the position of the top-left pixel can be controlled.
+During Mode 3, by default the PPU outputs one pixel to the screen per dot; the screen is 160 pixels wide, so the minimum Mode 3 length is 160 + 12[^first12] = 172 dots.
 
-Possible usage include a fixed status bar in an otherwise scrolling game (e.g.
-_Super Mario Land 2_).
+Unlike most game consoles, the Game Boy does not always output pixels steadily[^crt]: some features cause the rendering process to stall for a couple dots.
+Any extra time spent stalling *lengthens* Mode 3; but since scanlines last for a fixed number of dots, Mode 0 is therefore shortened by that same amount of time.
 
-### Objects
+Three things can cause Mode 3 "penalties":
 
-The background layer is useful for elements scrolling as a whole, but
-it's impractical for objects that need to move separately, such as the player.
+- **Background scrolling**: At the very beginning of Mode 3, rendering is paused for [`SCX`](<#FF42–FF43 — SCY, SCX: Viewport Y position, X position>) % 8 dots while the same number of pixels are discarded from the leftmost tile.
+- **Window**: After the last non-window pixel is emitted, a 6-dot penalty is incurred while the BG fetcher is being set up for the window.
+- **Objects**: Each object drawn during the scanline (even partially) incurs a 6- to 11-dot penalty ([see below](<#OBJ penalty algorithm>)).
 
-The _objects_ layer is designed to fill this gap: _objects_ are made of 1 or 2 stacked tiles (8×8 or 8×16 pixels)
-and can be displayed anywhere on the screen.
+On DMG and GBC in DMG mode, mid-scanline writes to [`BGP`](<#FF47 — BGP (Non-CGB Mode only): BG palette data>) allow observing this behavior precisely, as any delay shifts the write's effect to the left by that many dots.
 
-::: tip NOTE
+### OBJ penalty algorithm
 
-Several objects can be combined (they can be called _metasprites_) to draw
-a larger graphical element, usually called "sprite". Originally, the term "sprites"
-referred to fixed-sized objects composited together, by hardware, with a background.
-Use of the term has since become more general.
+Only the OBJ's leftmost pixel matters here, transparent or not; it is designated as "The Pixel" in the following.
 
-:::
+1. Determine the tile (background or window) that The Pixel is within. (This is affected by horizontal scrolling and/or the window!)
+2. If that tile has **not** been considered by a previous OBJ yet:
+   1. Count how many of that tile's pixels are to the right of The Pixel.
+   2. Subtract 3.
+   3. Incur this many dots of penalty, or zero if negative (from waiting for the BG fetch to finish).
+3. Incur a flat, 6-dot penalty (from fetching the OBJ's tile).
+
+**Exception**: an OBJ with an OAM X position of 0 (thus, completely off the left side of the screen) always incurs a 11-cycle penalty, regardless of `SCX`.
+
+TODO: a diagram of some examples would probably help this be much clearer!
```

Comment:
I think we should merge this PR as-is, and open an issue to get this resolved separately.

Reply:
Obviously, if we agree on that.

```suggestion
```